node/src/components/consensus/highway_core/state.rs

```diff
@@ -69,6 +69,29 @@ pub(crate) enum VoteError {
     NonLeaderBlock(ValidatorIndex),
     #[error("The vote is a block, but its parent is already a terminal block.")]
     ValueAfterTerminalBlock,
+    #[error("The vote's creator is banned.")]
+    Banned,
+}
+
+/// A reason for a validator to be marked as faulty.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub(crate) enum Fault<C: Context> {
+    /// The validator was known to be faulty from the beginning. All their messages are considered
+    /// invalid in this Highway instance.
+    Banned,
+    /// The validator is known to be faulty, but the evidence is not in this Highway instance.
+    Indirect,
+    /// We have evidence of the validator's fault.
```

Comment:
Can't `Banned` and `Indirect` be unified? Both indicate that validator is faulty b/c of equivocation in a past era.

Reply:
As discussed, `Banned` means that a validator was known to be malicious by everyone before the start of the era, so we can simply declare all of their votes invalid.

`Indirect` means we learned about their equivocation (in another era) while this era was running (or at least after the previous switch block), so we still have to consider their votes valid.

There is also a difference in priority: `Banned` can't be replaced by other fault types, but we replace `Indirect` with `Direct` evidence whenever we can, because that allows us to ignore the faulty validator in more future eras.